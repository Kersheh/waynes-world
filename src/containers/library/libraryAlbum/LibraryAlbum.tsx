import React from 'react';
import { useDispatch } from 'react-redux';

import Button from 'components/button/Button';
import { setAlbumView } from '../libraryActions';
import styles from './LibraryAlbum.module.scss';

interface LibraryAlbumProps {
  id: string;
  album: string;
  artist: string;
  // albumArt?: any;
}
const LibraryAlbum = ({ id, album, artist }: LibraryAlbumProps) => {
  const dispatch = useDispatch();

  return (
    <div className={styles.libraryAlbum}>
      <Button onClick={() => dispatch(setAlbumView(id))}>
        <div className={styles.albumArt} />

        <div className={styles.info}>
          <div className={styles.album}>{album}</div>
          <div className={styles.artist}>{artist}</div>
        </div>
      </Button>
    </div>
  );
};

export default LibraryAlbum;
