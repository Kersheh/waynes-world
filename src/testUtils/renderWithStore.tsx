import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose, Reducer } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { render } from '@testing-library/react';

import rootReducer from 'store/reducers';
import rootSaga from 'store/sagas';

const renderWithStore = (component: React.ReactElement) => {
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(
    rootReducer as Reducer,
    compose(applyMiddleware(sagaMiddleware))
  );
  sagaMiddleware.run(rootSaga);

  return render(<Provider store={store}>{component}</Provider>);
};

export default renderWithStore;
