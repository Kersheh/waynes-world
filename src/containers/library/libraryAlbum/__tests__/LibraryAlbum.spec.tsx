import React from 'react';

import renderWithStore from 'testUtils/renderWithStore';
import LibraryAlbum from '../LibraryAlbum';

describe('<LibraryAlbum/> component', () => {
  it('should render', () => {
    const { container } = renderWithStore(
      <LibraryAlbum
        id="some_album_id123"
        album="Album name"
        artist="Artist name"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
