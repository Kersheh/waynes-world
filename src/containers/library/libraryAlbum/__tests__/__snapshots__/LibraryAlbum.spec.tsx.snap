// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LibraryAlbum/> component should render 1`] = `
<div>
  <div
    class="libraryAlbum"
  >
    <button
      class="button text"
      type="button"
    >
      <div
        class="albumArt"
      />
      <div
        class="info"
      >
        <div
          class="album"
        >
          Album name
        </div>
        <div
          class="artist"
        >
          Artist name
        </div>
      </div>
    </button>
  </div>
</div>
`;
